service/controller/app/v1/values/configmap.go

```diff
@@ -0,0 +1,110 @@
+package values
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/giantswarm/apiextensions/pkg/apis/application/v1alpha1"
+	"github.com/giantswarm/microerror"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/app-operator/service/controller/app/v1/key"
+	appcatalogkey "github.com/giantswarm/app-operator/service/controller/appcatalog/v1/key"
+)
+
+func (v *Values) MergeConfigMapData(ctx context.Context, app v1alpha1.App, appCatalog v1alpha1.AppCatalog) (map[string]string, error) {
+	// We get the catalog level values if configured.
+	catalogData, err := v.getConfigMapForCatalog(ctx, appCatalog)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	// We get the app level values if configured.
+	appData, err := v.getConfigMapForApp(ctx, app)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	// Config is merged and in case of intersecting values the app level
+	// config is preferred.
+	mergedData, err := mergeConfigMapData(catalogData, appData)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	// We get the user level values if configured and merge them.
+	if key.UserConfigMapName(app) != "" {
+		userData, err := v.getUserConfigMapForApp(ctx, app)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+
+		// Config is merged again and in case of intersecting values the user
```

Comment:
Can we summarize this godocs and put it into the first place?

Reply:
Done.